StepIndex.md

```diff
@@ -0,0 +1,94 @@
+## Step Indices Issue
+
+Sequence steps are modeled in Scala with a simple ADT:
+
+````scala
+sealed abstract class Step[A] extends Product with Serializable {
+  def instrument: A
+}
+
+final case class BiasStep   [A](instrument: A)                             extends Step[A]
+final case class DarkStep   [A](instrument: A)                             extends Step[A]
+final case class GcalStep   [A](instrument: A, gcal:      GcalConfig)      extends Step[A]
+final case class ScienceStep[A](instrument: A, telescope: TelescopeConfig) extends Step[A]
+final case class SmartStep  [A](instrument: A, smartCal:  SmartCalConfig)  extends Step[A]
+````
+
+The database model mirrors this simple hierarchy with a "base" `step` table:
+
+````sql
+CREATE TABLE step (
+    observation_id character varying(40) NOT NULL,
+    index          smallint              NOT NULL,
+    instrument     identifier            NOT NULL,
+    step_type      step_type             NOT NULL
+);
+````
+
+and "sub" tables for the various step types:
+
+````sql
+CREATE TABLE step_science (
+    index          smallint              NOT NULL,
+    observation_id character varying(40) NOT NULL,
+    offset_p       double precision      NOT NULL,
+    offset_q       double precision      NOT NULL
+);
+````
+
+Here the primary key of each of the tables is the combination `(observation_id, index)`,
```

Comment:
Sorry, some parts are generated by `pgdump` which places the primary key definition elsewhere in the file:

````sql
ALTER TABLE ONLY step
    ADD CONSTRAINT step_pkey PRIMARY KEY (index, observation_id);
````